GoogleMeetup/src/main/webapp/script.js

```diff
@@ -315,6 +301,156 @@ function initAutocomplete() {
+        alert("You must add at least one person");
+        return false;
+    } if (destinationType == null) {
+        alert("You must select a destination type");
+        return false;
+    }
+    
+    return true;
+}
+
+
+// Renders the directions paths from the starting points to the meetup destination
+function drawPaths(){
+    var directionsService = new google.maps.DirectionsService();
+
+    for (var i = 0; i < attendees.length; i++) {
+        var request = {
+            origin: attendees[i].address,
+            destination: mapCenter,
+            travelMode: getTransitMode(attendees[i].transport)
+        };
+
+        var directionsRenderer = new google.maps.DirectionsRenderer();
+        directionsRenderer.setMap(map);
+        
+        directionsService.route(request, function(response, status) {
+            if (status === 'OK') {
+                directionsRenderer = new google.maps.DirectionsRenderer();
+                directionsRenderer.setMap(map);
+                directionsRenderer.setDirections(response);
+            }
+        });
+    }
+}
+
+
+// Returns the correct string for the directions API call
+function getTransitMode(transport) {
+    var transitModes = ["car", "public", "bike", "walk"];
+
+    console.assert(transitModes.includes(transport));
+
+    if (transport == "car") {
+        return "DRIVING";
+    } else if (transport == "public") {
+        return "TRANSIT";
+    } else if (transport == "bike") {
+        return "BICYCLING";
+    } else if (transport == "walk") {
+        return "WALKING";
+    }
+}
+
+
+// Converts the user inputted address into latLongs using the geocoder API
+var latLngs = [];
+function appendLatLang(attendee) {
+    var address = attendee['address'];
+    geocoder.geocode( { 'address': address}, function(results, status) {
+        if (status == 'OK') {
+            var latLng = results[0].geometry.location;
+            latLngs[latLngs.length] = latLng;
+
+            addAttendeeMarker();
+            centerMap();
+        } else { // For debugging
+            console.log('Geocode was not successful for the following reason: ' + status);
+        }
+    });
+}
+
+
+// Recenters the map based on the current mapCenter
+function centerMap() {
+    if (latLngs.length == 1) {
+        map.setCenter(latLngs[0]);
+        mapCenter = latLngs[0];
+    } else {
+        center = averageLatLongs();
+
+        map.setCenter(center);
+        mapCenter = center;
+    }
+}
+
+
+// Adds a new attendee marker to the map
+function addAttendeeMarker() {
+    var latLng = latLngs[latLngs.length - 1];
+    var marker = new google.maps.Marker({
+        map: map,
+        position: latLng
+    });
+}
+
+
+// Takes the average of the latLongs to find the approximate center of mass
+function averageLatLongs() {
+    var midpoint = {
+        "lat": latLngs[0].lat(), 
+        "lng": latLngs[0].lng()
+    };
+
+    for (i = 1; i < latLngs.length; i++) {
+        midpoint = {
```

Comment:
I don't think this is actually the center of mass. Consider applying this to latitudes 1, 2, 3. The average would be 2, but this algorithm produces ((1+2)/2+3)/2 = 3/4 + 3/2 = 9/4.

Reply:
Wait actually, true. I have no idea what I was doing there. I'll change it over to the actual average.

Reply:
Ok, now it is truly the average.